src/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeWriter.cs

```diff
@@ -210,7 +210,7 @@ public override void Complete(Exception exception = null)
 
             _isCompleted = true;
 
-            FlushInternal();
+            FlushInternal(writeToStream: exception == null);
```

Comment:
Maybe we should set this unconditionally to `false`?

Reply:
That is: the user is responsible for flushing by using `PipeWriter.FlushAsync` and `PipeWriter.WriteAsync`. No implicit flushing at Complete.

Reply:
I didn't realize this exception isn't propagated. So as it is, setting the exception or not acts as a boolean that controls flushing.